code/game/machinery/computer/arcade.dm

```diff
@@ -50,6 +50,9 @@
 		qdel(src)
 
 /obj/machinery/computer/arcade/proc/prizevend()
+	if(prob(prob(prob(1)))) //1 in a million
+		new /obj/item/weapon/gun/energy/pulse(contents)
+
```

Comment:
Actually I was wrong, adding a return will just spawn the gun in the arcade's contents.

Letting it continue will then dispense it as normal.